src/metpy/plots/patheffects.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (c) 2016,2017,2019 MetPy Developers.
+# Distributed under the terms of the BSD 3-Clause License.
+# SPDX-License-Identifier: BSD-3-Clause
+"""Classes for matplotlib path effects."""
+
+from matplotlib.path import Path
+from matplotlib.patheffects import _subclass_with_normal, AbstractPathEffect
+import numpy as np
+
+from ..package_tools import Exporter
+
+exporter = Exporter(globals())
+
+
+@exporter.export
+class ScallopedStroke(AbstractPathEffect):
+    """A line-based PathEffect which draws a path with a scalloped style.
+
+    The spacing, length, and side of the scallops can be controlled. This implementation is
+    based off of :class:`matplotlib.patheffects.TickedStroke`.
+    """
+
+    def __init__(self, offset=(0, 0), spacing=10.0, side='left', length=1.15, **kwargs):
+        """Create a scalloped path effect.
+
+        Parameters
+        ----------
+        offset : (float, float), default: (0, 0)
+            The (x, y) offset to apply to the path, in points.
+        spacing : float, default: 10.0
+            The spacing between ticks in points.
+        side : str, default: left
+            Side of the path scallops appear on from the reference of
+            walking along the curve. Options are left and right.
+        length : float, default: 1.414
+            The length of the tick relative to spacing.
+        kwargs :
+            Extra keywords are stored and passed through to
+            :meth:`AbstractPathEffect._update_gc`.
+        """
+        super().__init__(offset)
+
+        self._spacing = spacing
+        if side == 'left':
+            self._angle = 90
+        elif side == 'right':
+            self._angle = -90
+        else:
+            raise ValueError('Side must be left or right.')
+        self._length = length
+        self._gc = kwargs
+
+    def draw_path(self, renderer, gc, tpath, affine, rgbFace):
+        """Draw the path with updated gc."""
+        # Do not modify the input! Use copy instead.
+        gc0 = renderer.new_gc()
+        gc0.copy_properties(gc)
+
+        gc0 = self._update_gc(gc0, self._gc)
+        try:
+            # For matplotlib >= 3.2
+            trans = affine + self._offset_transform(renderer)
+        except TypeError:
+            # For matplotlib < 3.2
+            trans = self._offset_transform(renderer, affine)
+
+        theta = -np.radians(self._angle)
+        trans_matrix = np.array([[np.cos(theta), -np.sin(theta)],
+                                 [np.sin(theta), np.cos(theta)]])
+
+        # Convert spacing parameter to pixels.
+        spacing_px = renderer.points_to_pixels(self._spacing)
```

Comment:
Does this "blow up" when using a PS/SVG backend that does not have the concept of pixels?

Reply:
Thanks and sorry, I was vague.  I was wondering if passing a points value of say `30` would cause the rendered path on SVG to look wildly bad.  I've personally struggled with an issue like this attempting to figure out how something will look within matplotlib figure before the renderer is called/known.

Reply:
I haven't tried it on this PR, but it's working great on #2420 with the same call.

Reply:
OK, thank you!  This all is very cool and thanks @nawendt !